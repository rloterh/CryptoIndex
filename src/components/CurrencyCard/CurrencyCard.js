import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import { ListGroup } from 'react-bootstrap';
import CurrencyImg from '../../assets/currency.png';
import './CurrencyCard.scss';

const CurrencyCard = (props) => {
  const {
    id, name, symbol, priceUsd, priceBtc, mC, tSp, rank,
  } = props;
  const navigate = useNavigate();
  const goToDetails = () => {
    navigate('details', {
      state: {
        id, name, symbol, priceUsd, priceBtc, mC, tSp, rank,
      },
    });
  };
  const altCurency = { style: 'currency', currency: 'USD' };
  const convertToDollars = new Intl.NumberFormat('en-US', altCurency);
  return (
    <ListGroup.Item key={id} onClick={goToDetails} data-testid="list-group-item">
      <div className="d-flex" id="data-container">
        <div className="bottom-data d-flex">
          <img src={CurrencyImg} alt="currency-icon" className="currency-image" />
          <i className="far fa-arrow-alt-circle-right" />
        </div>
        <h3>{name}</h3>
        <p>
          {' ( '}
          {symbol}
          {' ) '}
        </p>
        <p>
          { convertToDollars.format(priceUsd) }
        </p>
      </div>
    </ListGroup.Item>
  );
};

CurrencyCard.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  priceUsd: PropTypes.string.isRequired,
  priceBtc: PropTypes.string.isRequired,
  mC: PropTypes.string.isRequired,
  tSp: PropTypes.string.isRequired,
  rank: PropTypes.number.isRequired,
};

export default CurrencyCard;
