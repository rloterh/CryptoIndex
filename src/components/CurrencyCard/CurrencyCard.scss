
.list-group-item:nth-child(4n + 1),
.list-group-item:nth-child(4n + 4) {
  background-color: #ffc843;
}

.list-group-item:nth-child(4n + 2),
.list-group-item:nth-child(4n + 3) {
  background-color: #d38700;
}

#data-container {
  justify-content: flex-end;
  flex-direction: column;
  padding: 1vh;
  padding-top: 2vh;
  font-family: 'Roboto', sans-serif;
  h3 {
    font-size: 1.2rem;
    font-weight: bold;
    text-align: right;
  };
  p {
    font-size: 0.88rem;
    margin-bottom: 0.7vh;
    text-align: right;
  }
}

.list-group-item {
  background-color: transparent;
  color: #fff;
  display: flex;
  justify-content: flex-start;
  padding-top: 2vh;
  flex: 1;
  min-width: 50%;
  border: 0;
  cursor: pointer;
}

.eth-icon {
  height: 10vh;
  width: 10vh;
  margin-bottom: 2vh;
}

.bottom-data > i {
  margin-left: 15vw;
}
