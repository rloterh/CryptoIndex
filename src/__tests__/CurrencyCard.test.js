import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import * as reactRedux from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CurrencyCard from '../components/CurrencyCard/CurrencyCard';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

describe('CurrencyCard', () => {
  const useSelectorMock = reactRedux.useSelector;
  const useDispatchMock = reactRedux.useDispatch;
  const mockStore = {
    currencyReducer: [
      {
        id: '1',
        name: 'BitCoin-Test',
        symbol: 'BTC-TEST',
        priceUsd: '100',
        priceBtc: '1',
        marketCap: '32000',
        totalSupply: '333',
        rank: 1,
      },
    ],
  };
  beforeEach(() => {
    useDispatchMock.mockImplementation(() => () => {});
    useSelectorMock.mockImplementation((selector) => selector(mockStore));
  });
  afterEach(() => {
    useDispatchMock.mockClear();
    useSelectorMock.mockClear();
  });

  test('Create and show list of Currencies', () => {
    render(<CurrencyCard />);
    expect(screen.getByTestId('list-group-item')).toBeInTheDocument();
  });
  test('Create CurrencyCard Component inside Rockets List', () => {
    const currencyData = useSelectorMock((mockStore) => mockStore.currencyReducer[0]);
    render(
      <CurrencyCard
        key={currencyData.id}
        id={currencyData.id}
        name={currencyData.name}
        symbol={currencyData.symbol}
        priceUsd={currencyData.priceUsd}
        priceBtc={currencyData.priceBtc}
        mC={currencyData.marketCap}
        tSp={currencyData.totalSupply}
        rank={currencyData.rank}
      />,
    );
    expect(screen.getByTestId('list-group-item')).toBeInTheDocument();
  });
});

describe('checks if CurrencyCard component renders correctly', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <CurrencyCard />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
