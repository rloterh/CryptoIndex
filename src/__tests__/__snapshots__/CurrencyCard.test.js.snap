// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if CurrencyCard component renders correctly should render correctly 1`] = `
<div
  className="list-group-item"
  data-testid="list-group-item"
  onClick={[Function]}
>
  <div
    className="d-flex"
    id="data-container"
  >
    <div
      className="bottom-data d-flex"
    >
      <img
        alt="currency-icon"
        className="currency-image"
        src="currency.png"
      />
      <i
        className="far fa-arrow-alt-circle-right"
      />
    </div>
    <h3 />
    <p>
       ( 
       ) 
    </p>
    <p>
      $NaN
    </p>
  </div>
</div>
`;
